tmt/steps/report/html.py

```diff
@@ -58,21 +58,33 @@ def go(self) -> None:
 
         # Prepare the template
         environment = jinja2.Environment()
-        environment.filters["basename"] = lambda x: os.path.basename(x)
+        environment.filters["basename"] = lambda x: Path(x).name
 
         if self.get('absolute-paths'):
-            environment.filters["linkable_path"] = os.path.abspath
+            environment.filters["linkable_path"] = lambda x: str(Path(x).absolute())
         else:
             # Links used in html should be relative to a workdir
-            environment.filters["linkable_path"] = lambda x: os.path.relpath(x, self.workdir)
+            def relpath(_path: str) -> str:
+                path = Path(_path)
+
+                assert self.workdir is not None  # narrow type
+                if tmt.utils.is_relative_to(path, self.workdir):
+                    return str(path.relative_to(self.workdir))
+
+                # Apparently, pathlib does not offer relpath transition
+                # between parallel trees, we have to use the "old" os.path
+                # package.
+                return os.path.relpath(_path, str(self.workdir))
```

Comment:
Seems a bit complicated. If `pathlib` does not provide an elegant solution for relative paths why not just keep the `os.path.relpath` one-liner as it was?

Reply:
Trying out the custom `Path`  way in 9275aeabe66a5bcd3e8d57d70bbcb428c9ac2957 - if it works, it seems to me easier to work with.

Reply:
Yes, this looks much more clear & tidy. Thanks!